boards/nordic/nrf52840_dongle/src/io.rs

```diff
@@ -25,7 +25,10 @@ impl Write for Writer {
 
 impl IoWrite for Writer {
     fn write(&mut self, buf: &[u8]) {
-        let uart = unsafe { &mut nrf52840::uart::UARTE0 };
+        // Here, we create a second instance of the Uarte struct.
+        // This is okay because we only call this during a panic, and
+        // we will never actually process the interrupts
+        let uart = nrf52840::uart::Uarte::new();
```

Comment:
Does this not require `unsafe`?

Reply:
currently no. We could consider making it take a 'PeripheralCreationCapability'.